### PR TITLE
feat: collections

### LINKED ISSUE #5166: [organization] stacks and folders are just collections
- underlying both stacks & folders are collections of objects
- stack is just one way to render a collection of objects (another would be grid)
- we cannot simply render a folder as a stack/grid though as they have extra view metadata
  - this metadata could have reasonable defaults, but if edited it needs to be persisted somewhere
- the expectation is to resolve this with a "data layer" above the ECHO db where different objects can be joined together into a single view
- with resolvers in a data layer a "stack" can be a joined view of a collection and stack view metadata (e.g. section heights)
- currently blocked by #5124 as that is what will form the data layer (which should probably be another issue as well)

### LINKED ISSUE #6104: [organization] make spaces look like collections

### LINKED ISSUE #6166: [stack] show document title in collapsed section
image

image

### PR DESCRIPTION
**Pull request by wittjosiah**
### Motivation / Background

This Pull Request introduces the concept of collections, replacing folders and stacks at the data level.

Resolves #5166
Resolves #6166
Resolves #6104 
Unblocks #4886

### Details

- removes the old folder & stack types from `@braneframe/types`
- adds a `Collection` type
- adds a `StackView` type
- removes graph builder & intent resolver from stack plugin, updates it to learn how to render collections
- consolidates extra view data regarding sections (e.g. collapsed) into the section type to be wired up to mosaic tiles as items
- renders spaces as the root collection

### Additional information

- the design of the data structure ensures that the state of a view always converges, however given that views are referenced objects there could end up being orphaned objects. Consider whether we should clean these up or perhaps if views shouldn't be separate objects.
- notably this does not yet allow objects to exist in multiple collections at once (at the app level)
- allowing objects to exist in multiple collections will follow but requires some graph work as well as UX work
- it would be good to wire up a file system directory handle to map to the collection schema and render as a stack too but that can also be done as a followup

TODO
- [x] migrations
- [x] tests
- [x] verify presentations still work
- [ ] verify labs functions still work
- [x] review remaining "folder" references

image

### CODE REVIEW

File: packages/apps/types/src/schema/collection.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { Expando, ref, S, TypedObject } from '@dxos/echo-schema';
+
+export class Collection extends TypedObject({ typename: 'dxos.Collection', version: '0.1.0' })({
+  name: S.optional(S.string),
+  objects: S.mutable(S.array(ref(Expando))),
+  // Key is schema typename and value is reference to a view object of the associated schema.
+  // Having collection reference the views rather than vice versa ensures that the state converges to a single view per key (i.e. type).
+  // This also leaves open a future where this key could be changed to allow for multiple stack views per section.
+  // TODO(wittjosiah): Any way to make this more type safe?
+  views: S.mutable(S.record(S.string, ref(Expando))),
```
Comment (thure) [inline packages/apps/types/src/schema/collection.ts:14]: What kinds of objects can `views` reference?

Comment (thure) [reply packages/apps/types/src/schema/collection.ts:14]: At first glance, this seems like it should be much less permissive than `objects`.

File: packages/ui/react-ui-stack/src/components/Section.tsx [hunk 2]
```diff
@@ -299,70 +306,53 @@ export const SectionToolbar = ({ children, classNames }: SectionToolbarProps) =>
...
-      </Section>
-    );
+  const section = (
+    <Section
+      ref={forwardedRef}
+      title={title}
+      id={transformedItem.id}
+      size={transformedItem.view?.size}
+      icon={transformedItem.metadata?.icon}
+      collapsed={transformedItem.view?.collapsed}
```
Comment (thure) [inline packages/ui/react-ui-stack/src/components/Section.tsx:341]: I see, this is making sense and looking intuitive 👍

File: packages/apps/types/src/schema/collection.ts (hunk 1 above)
Comment (wittjosiah) [reply packages/apps/types/src/schema/collection.ts:14]: The object for the view should be of the schema associated with the `typename` of the record key. Would be nice to find a way to have that be more strongly typed but I didn't want to block on that. Open to ideas here though

**Pull request merged**